iohk-monitoring/test/Cardano/BM/Test/Monitoring.lhs

```diff
@@ -173,3 +195,80 @@ parseEvalExpression t res env =
         Just e  -> evaluate env e @?= res
 
 \end{code}
+
+\subsubsection{Actions tests}
+
+\begin{code}
+monitoringThr :: Trace IO Text -> IO (Async.Async ())
+monitoringThr trace = do
+    trace' <- appendName "monitoring" trace
+    proc <- Async.async $ sendTo trace'
+    return proc
+  where
+    sendTo tr = do
+        (,) <$> (mkLOMeta Warning Public)
+            <*> pure (LogValue "monitMe" (PureI 100))
+        >>= traceNamedObject tr
+
+testSetGlobalMinimalSeverity :: Assertion
+testSetGlobalMinimalSeverity = do
+    let initialGlobalSeverity = Debug
+        targetGlobalSeverity  = Info
+
+    c <- CM.empty
+    CM.setMinSeverity c initialGlobalSeverity
+    CM.setDefaultBackends c [MonitoringBK]
+    CM.setSetupBackends c [MonitoringBK]
+
+    CM.setBackends c "complex.monitoring.monitMe" (Just [MonitoringBK])
+
+    CM.setMonitors c $ HM.fromList
+        [ ( "complex.monitoring"
+          , ( Nothing
+            , Compare "monitMe" (GE, (OpMeasurable 10))
+            , [SetGlobalMinimalSeverity targetGlobalSeverity]
+            )
+          )
+        ]
+
+    (tr' :: Trace IO Text, _) <- setupTrace_ c "complex"
+
+    procMonitoring <- monitoringThr tr'
+    _ <- Async.waitCatch procMonitoring
+
+    threadDelay 1000000  -- 1 second
+    currentGlobalSeverity <- CM.minSeverity c
+    assertBool "Global minimal severity didn't change!" $
+        currentGlobalSeverity == targetGlobalSeverity
+
+testAlterSeverity :: Assertion
+testAlterSeverity = do
+    let initialSeverity = Debug
+        targetSeverity  = Info
+
+    c <- CM.empty
+    CM.setSeverity c "complex.monitoring.monitMe" (Just initialSeverity)
+    CM.setDefaultBackends c [MonitoringBK]
+    CM.setSetupBackends c [MonitoringBK]
+
+    CM.setBackends c "complex.monitoring.monitMe" (Just [MonitoringBK])
+
+    CM.setMonitors c $ HM.fromList
+        [ ( "complex.monitoring"
+          , ( Nothing
+            , Compare "monitMe" (GE, (OpMeasurable 10))
+            , [AlterSeverity "complex.monitoring.monitMe" targetSeverity]
+            )
+          )
+        ]
+
+    (tr' :: Trace IO Text, _) <- setupTrace_ c "complex"
+
+    procMonitoring <- monitoringThr tr'
+    _ <- Async.waitCatch procMonitoring
+
+    threadDelay 1000000  -- 1 second
```

Comment:
why do you need to wait here for a second?

Reply:
Experimentally... :-) I copied that code from `example-complex`, and when I removed that 1-second pause - test fails.